library/src/main/scala/scala/scalajs/js/BigInt.scala

```diff
@@ -0,0 +1,188 @@
+package scala.scalajs.js
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation.JSGlobal
+
+/**
+ * A built-in object that provides a way to represent whole numbers larger than
+ * 2 ^ 53 - 1, which is the largest number JavaScript can reliably represent
+ * with the Number primitive.
+ *
+ * BigInt can be used for arbitrarily large integers.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt
+ */
+@js.native
+@JSGlobal
+class BigInt private[this] () extends js.Object {
+
+  def +(other: BigInt): BigInt = js.native
+  def *(other: BigInt): BigInt = js.native
+  def -(other: BigInt): BigInt = js.native
+  def **(other: BigInt): BigInt = js.native
+  def %(other: BigInt): BigInt = js.native
+
+  def &(other: BigInt): BigInt = js.native
+  def |(other: BigInt): BigInt = js.native
+  def ^(other: BigInt): BigInt = js.native
+  def <<(other: BigInt): BigInt = js.native
+  def >>(other: BigInt): BigInt = js.native
+  def >>>(other: BigInt): BigInt = js.native
+  // no >>>
+
+  def unary_- : BigInt = js.native
+  def unary_~ : BigInt = js.native
+  // no unary_+
+
+  def ==(x: BigInt): Boolean = js.native
+  def ==(x: Short): Boolean = js.native
+  def ==(x: Char): Boolean = js.native
+  def ==(x: Int): Boolean = js.native
+  def ==(x: Long): Boolean = js.native
+  def ==(x: Float): Boolean = js.native
+  def ==(x: Double): Boolean = js.native
```

Comment:
Overloading `==` and `!=` is generally considered bad practice. I don't think there is a strong reason to do it here, so I would remove the overloads.